File: solutions/1261/1261-kir3i.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class FindElements {
public:
    unordered_set<int> s;
    FindElements(TreeNode* root) {
        setVal(root, 0);
    }
    
    void setVal(TreeNode *node, const int &val) {
        if(!node)   return;
        node->val = val;
        s.insert(val);
        setVal(node->left, 2*val+1);
        setVal(node->right, 2*val+2);
    }
    
    bool find(int target) {
        return s.find(target) != s.end();
    }
};

File: solutions/1261/1261-kir3i.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class FindElements:
    
    def __init__(self, root):
        self.s = set([])
        self.setVal(root, 0)
    def setVal(self, node, val):
        if not node:
            return
        node.val = val
        self.s.add(val)
        self.setVal(node.left, 2*val+1)
        self.setVal(node.right, 2*val+2)

    def find(self, target):
        if target in self.s:
            return True
        else:
            return False